### PR TITLE
Fixed MM-33249 issue

### PR DESCRIPTION
**Pull request by manojmalik20**
#### Summary
Fixed MM-33249 Mobile > Unable to set custom status emoji via slash command

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33249

#### Release Note
```release-note
Added the logic for detecting Unicode emoji and setting it as custom status emoji via slash commands.
```

### CODE REVIEW

**Review by hmhealey — CHANGES_REQUESTED** [review]
The code looks good, but would you mind adding some unit tests for the emoji detection logic? I think it would be good to have tests for the Unicode case as well as some more difficult cases like multi-part family emojis and messages containing Unicode emojis. Perhaps you could pull out the whole block for constructing the `CustomStatus` into its own function to simplify that testing since it would just take a string and return a `CustomStatus`.

**Comment by manojmalik20** [comment]
While going through the SystemEmojis map in `emojiData.go`, I noticed that multiple emoji names contain same unicode. For eg-
`man_singer_medium_skin_tone`, `man_teacher_medium_skin_tone`, `family_man_woman_girl_boy_medium_skin_tone`.. and many more have the same unicode `1f468-1f3fd`... so does it mean that the SystemEmojis map is wrong?

**Comment by hmhealey** [comment]
That is a very good find, and it's definitely a problem. The existing code doesn't actually look at those unicode values (just whether or not the emoji exists in the map), so it's never caused a problem before.

That `1f468-1f3fd` one you mentioned is "man light skin tone", so that's the only one that should have that file name. Looking at the data files in the web app, they also have the same issue where the filename is duplicated. Posting `:man_farmer_light_skin_tone:` in the web app also outputs the wrong emoji (`:man_light_skin_tone:`), so the issue is bigger than just the server's data files. I'll file a ticket to investigate that further. In the mean time, there are definitely some correct emojis with skin tones, so maybe we can find some to use for tests while we fix the data files?

Also, CCing @ethervoid since he added the generation for that file in https://github.com/mattermost/mattermost-webapp/pull/3429 in case he remembers anything about this issue. There is a comment in the `make-emojis` script about that, so that PR didn't cause the issue though.

**Comment by ethervoid** [comment]
@hmhealey uf no idea, this was my first PR for MM so not a clue sorry

**Comment by manojmalik20** [comment]
@hmhealey Can you rerun the CI? I've fixed the lint errors.

**Comment by hmhealey** [comment]
@ethervoid No worries. The problem seems to predate your changes since the issue appears in the webapp's emoji data as well, and I figured it was a long shot that you would remember it from that far back.

@manojmalik20 I've filed a ticket to look into the incorrect unicode values being generated for those ones. Going a bit more technical, it seems to only affect certain "job" emojis using both a skin tone modifier and a ZWJ sequence (farmer, teacher, astronaut, etc). I filed [MM-35234](https://mattermost.atlassian.net/browse/MM-35234) to look into that, but in the mean time, I wonder if there's a good way to work around that in the mean time.

Perhaps we can identify if an emoji's unicode looks like `<man/woman>_<skin tone>_200d_` and remove the skin tone modifier for the time being. It looks like since the unicode is messed up for those emojis, we're not actually generating images for the skin tone variations, so we'll have to fall back to the generic skin tone on those. That pattern should catch all the ones we know about while we fix the data file generator.

Also, that family emoji with a skin tone modifier you mentioned isn't supported. From what I can tell, families with skin tone modifiers haven't made it into an official emoji spec, so we can skip that and just test the ones with the default skin tone.

**Comment by manojmalik20** [comment]
> Perhaps we can identify if an emoji's unicode looks like `<man/woman>_<skin tone>_200d_` and remove the skin tone modifier for the time being.
@hmhealey How do you propose to implement this?

**Comment by hmhealey** [comment]
Once you have the unicode as a hex string, you could try a regex to match that. I think `^(1f468|1f469)_(1f3fb|1f3fc|1f3fd|1f3fe|1f3ff)_200d_` might do it. The first group is man/woman and the second is all the skin tones.

**Comment by manojmalik20** [comment]
@hmhealey I didn't use `^(1f468|1f469)` because it doesn't cover certain emojis like `basketball_man`, `surfing_man` etc.. 
This code works with almost all emojis except some family emojis with skin tones. You can review and test it yourself.

**Review by hmhealey — COMMENTED** [review]
Maybe instead of immediately removing the skin tone, maybe we could check if we have the emoji with a skin tone first removing it? That'll give us some nice forwards compatibility when we eventually support the skin tones for those emojis.

**Review by hmhealey — APPROVED** [review]
Looks good to me!

**Review by isacikgoz — COMMENTED** [review]
Looks good to me overall, have some styling concerns.

File: app/slashcommands/command_custom_status.go [hunk 1]
```diff
@@ -58,28 +60,74 @@ func (*CustomStatusProvider) DoCommand(a *app.App, args *model.CommandArgs, mess
...
-		ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
-		Text: args.T("api.command_custom_status.success", map[string]interface{}{
-			"EmojiName":     ":" + customStatus.Emoji + ":",
-			"StatusMessage": customStatus.Text,
-		}),
+func removeUnicodeSkinTone(unicodeString string) string {
+	skinToneDetectorRegex := regexp.MustCompile("-(1f3fb|1f3fc|1f3fd|1f3fe|1f3ff)")
+	skinToneLocations := skinToneDetectorRegex.FindIndex([]byte(unicodeString))
+
+	if len(skinToneLocations) > 0 {
```
Comment (isacikgoz) [inline app/slashcommands/command_custom_status.go:119]: Same indentation case here.

**Review by ogi-m — APPROVED** [review]
LGTM 👍🏻

**Comment by manojmalik20** [comment]
@esethna Can you please add more reviewers or the `EETests` label in this PR?

**Comment by esethna** [comment]
@isacikgoz can you please help with a re-review and approve and merge if your concerns have been addressed? 

@ogi-m I'm not sure how the EETests label is supposed to work? Can you help here as needed? ^

**Review by isacikgoz — APPROVED** [review]
